onnxruntime/core/mlas/lib/amd64/QLinearBinaryOpKernelAvx2.asm

```diff
@@ -0,0 +1,436 @@
+;++
+;
+; Copyright (c) Microsoft Corporation. All rights reserved.
+;
+; Licensed under the MIT License.
+;
+; Module Name:
+;
+;    QLinearBinaryOpKernelAvx2.asm
+;
+; Abstract:
+;
+;    This module implements the kernels for the quantized linear add
+;    for element type int8_t and uint8_t.
+;
+;    This implementation uses AVX2 instructions.
+;
+;--
+
+        .xlist
+INCLUDE mlasi.inc
+        .list
+
+        EXTERN  MLasPackBytesMM256VpshufbControl:NEAR
+        EXTERN  MLasPackBytesMM256VpermpsControl:NEAR
+
+;
+; Stack frame layout for the QLinearBinaryOp kernels.
+;
+
+QLinearBinaryOpFrame STRUCT
+        SavedXmm6 OWORD ?
+        SavedXmm7 OWORD ?
+        SavedXmm8 OWORD ?
+        SavedXmm9 OWORD ?
+        SavedXmm10 OWORD ?
+        SavedXmm11 OWORD ?
+        Padding0 QWORD ?
+        Padding1 QWORD ?
+        Padding2 QWORD ?
+        ReturnAddress QWORD ?
+        PreviousP1Home QWORD ?
+        PreviousP2Home QWORD ?
+        PreviousP3Home QWORD ?
+        PreviousP4Home QWORD ?
+        ScaleB  QWORD ?
+        ZeroPointB QWORD ?
+        ScaleC QWORD ?
+        ZeroPointC QWORD ?
+        OutputC QWORD ?
+        LengthA QWORD ?
+        LengthB QWord ?
+QLinearBinaryOpFrame ENDS
+
+;
+; Macro Description:
+;
+;   This macro generates code to extend signed/unsigned int8 to
+;   signed/unsigned int64/32/16 respectively, according to DataType
+;   and target.
+;
+; Arguments:
+;
+;   Target - target register, could be 16/32/64 bits one.
+;
+;   Source - Supplies address of 8 x 8bits integers.
+;
+;   DataType - S8 or U8
+;
+
+Extend8BitsInt MACRO Target, Source, DataType
```

Comment:
consider doing Extend8BitsIntU8 and Extend8BitsIntS8 macros and then have your other kernel just do Extend8BitsInt&DataType to resolve the right macro. Up to you...

Reply:
Sure! Try different thinking helps.